deploy.go

```diff
@@ -4,51 +4,107 @@
 
 package main
 
-import "log"
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
 
-func build(c Client, appName string, cmd []string) {
-	log.SetFlags(0)
-	envs, err := c.getAppEnvs(appName)
-	if err != nil {
-		log.Fatal(err)
-	}
-	err = execScript(cmd, envs, nil)
+	"github.com/tsuru/deploy-agent/internal/docker"
+	"github.com/tsuru/deploy-agent/internal/tsuru"
+	"github.com/tsuru/tsuru/exec"
+)
+
+func build(c tsuru.Client, appName string, cmd []string, fs Filesystem, executor exec.Executor) error {
+	envs, err := c.GetAppEnvs(appName)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
+	return execScript(cmd, envs, os.Stdout, fs, executor)
 }
 
-func deploy(c Client, appName string) {
-	log.SetFlags(0)
-	var yamlData TsuruYaml
-	envs, err := c.registerUnit(appName, yamlData)
+func deploy(c tsuru.Client, appName string, fs Filesystem, executor exec.Executor) error {
+	var yamlData tsuru.TsuruYaml
+	envs, err := c.RegisterUnit(appName, yamlData)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
-	diff, firstDeploy, err := readDiffDeploy()
-	if err != nil {
-		log.Fatal(err)
-	}
-	if !firstDeploy {
-		err = c.sendDiffDeploy(diff, appName)
+	diff, firstDeploy, err := readDiffDeploy(fs)
+	if !firstDeploy && err == nil {
```

Comment:
@cezarsa whats your take on this one? is it safe to continue?

Reply:
I really don't like how we left the container diff half implemented with pieces all around. This serves no useful purpose for now as it needs the `$EXPERIMENTAL_DIFF_ENABLED` env to be set in the base-platform and we never set it anywhere, also as we don't test it it's probably broken by now. So, I'm in favor of removing this Diff code from here.

If we are to support diffs again in the future I think we should do it from the outside, calculating the diff based on the app image layers using something like https://github.com/GoogleCloudPlatform/container-diff to do the heavy lifting.

Reply:
Sounds good, I'll remove it completely in a forthcoming PR.